crates/turborepo-lib/src/run/mod.rs

```diff
@@ -233,6 +213,43 @@ impl<'a> Run<'a> {
             None
         };
 
+        let mut pkg_dep_graph =
+            PackageGraph::builder(&self.base.repo_root, root_package_json.clone())
+                .with_single_package_mode(opts.run_opts.single_package)
+                .with_package_discovery(FallbackPackageDiscovery::new(
+                    daemon.as_mut().map(DaemonPackageDiscovery::new),
+                    LocalPackageDiscoveryBuilder::new(
+                        self.base.repo_root.clone(),
+                        None,
+                        Some(root_package_json.clone()),
+                    )
+                    .build()?,
```

Comment:
Shouldn't be changed in this PR as we want to get this merged, but maybe a leave a `TODO` to allow for a `FallbackPackageDiscovery` to take a `PackageDiscoveryBuilder` as a fallback strategy that only gets built if needed.